### PR TITLE
Initial version of nsntrace snap

### PR DESCRIPTION
**Pull request by jonasdn**
This YAML can be used by snapcraft to create a snap package image of
nsntrace.

### CODE REVIEW

**Review by keachi — APPROVED** [review]

**Comment by pabs3** [comment]
The stage-packages looks a bit weird, but I guess snap doesn't have the automatic dependency calculation things that regular distros like Debian have.

**Review by pabs3 — APPROVED** [review]

**Comment by keachi** [comment]
Perhaps using a stable release instead of the "current" git commit would be a good idea. Does snap support versioning?
Suggestion: release snap version 3 with [project release 3 code](https://github.com/nsntrace/nsntrace/releases/tag/v3).

**Comment by jonasdn** [comment]
> Perhaps using a stable release instead of the "current" git commit would be a good idea. Does snap support versioning?
> Suggestion: release snap version 3 with [project release 3 code](https://github.com/nsntrace/nsntrace/releases/tag/v3).

I am a bit new at snapcraft, so I hope I got this correctly. I updated the PR to have the snapcraft YAML tell snapcraft to always fetch the latest _tag_ and use that for building and versioning.

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+grade: stable
+confinement: classic
+
+parts:
+  nsntrace:
+    plugin: autotools
+    source-type: git
+    override-pull: | # This override tells snapcraft to always build and version using latest tag
+      snapcraftctl pull
+      last_tag="$(git describe --tags --abbrev=0)"
```
Comment (mattiasb) [inline snap/snapcraft.yaml:17]: Just a drive-by comment (not even knowing what nsntrace is :D):
You might want to add a `--match 'v*'` here to avoid using potential tags that aren't releases.

Something like this:
```
last_tag="$(git describe --tags --abbrev=0 --match 'v*')"
```

Comment (jonasdn) [reply snap/snapcraft.yaml:17]: Sounds lika a good idea! Thanks for driving by.

**Comment by jonasdn** [comment]
> The stage-packages looks a bit weird, but I guess snap doesn't have the automatic dependency calculation things that regular distros like Debian have.

The snapcraft tool told me to add it, I obeyed. : )
